packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -605,22 +574,14 @@ export function getWorkflowHooksMain(
 	data: IWorkflowExecutionDataProcess,
 	executionId: string,
 ): WorkflowHooks {
-	const hookFunctions = hookFunctionsSave();
-	const pushFunctions = hookFunctionsPush();
-	for (const key of Object.keys(pushFunctions)) {
-		const hooks = hookFunctions[key] ?? [];
-		hooks.push.apply(hookFunctions[key], pushFunctions[key]);
-	}
-
-	const preExecuteFunctions = hookFunctionsPreExecute();
-	for (const key of Object.keys(preExecuteFunctions)) {
-		const hooks = hookFunctions[key] ?? [];
-		hooks.push.apply(hookFunctions[key], preExecuteFunctions[key]);
-	}
-
-	if (!hookFunctions.nodeExecuteBefore) hookFunctions.nodeExecuteBefore = [];
-	if (!hookFunctions.nodeExecuteAfter) hookFunctions.nodeExecuteAfter = [];
-
+	const hookFunctions = mergeHookFunctions(
+		hookFunctionsWorkflowEvents(),
+		hookFunctionsNodeEvents(),
+		hookFunctionsFinalizeExecutionStatus(),
+		hookFunctionsSave(),
+		hookFunctionsPush(),
+		hookFunctionsPreExecute(),
```

Comment:
A bit odd that `hookFunctionsPreExecute` comes at the very end. Then again it contains `workflowExecuteBefore` and `nodeExecuteAfter` so we might have to keep like this for now.

Reply:
`hookFunctionsPreExecute` will soon be deleted, and will be replaced with other functions that make a lot more sense.